### PR TITLE
Migration file created and login api made 

### LINKED ISSUE #9: Add Login API
### Problem Description
So far, backend needs a login api to break start for both backend and frontend

### Solution Proposed
api endpoint: /login
http method: post
parameters: email, password

### PR DESCRIPTION
**Pull request by surazad99**
All the relationships among tables created 
User Table Seeder made 
Login Api made with oAuth token using passport laravel package 
Fixes #9 
Part of #10

### CODE REVIEW

File: backend/app/Http/Controllers/Api/LoginController.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+
+class LoginController extends Controller
+{
+    public function login(Request $request){
+
+        $login = $request->validate([
+            'email' => 'required|string',
```
Comment (kiranparajuli589) [inline backend/app/Http/Controllers/Api/LoginController.php:14]:
```suggestion
            'email' => 'required|email|string',
```

**Comment by kiranparajuli589** [comment]
How is user-password strength managed?

File: backend/app/Http/Controllers/Api/LoginController.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            'password' => 'required|string'
+        ]);
+
+        if(!Auth::attempt($login)){
+            return response(['message' => 'invalid Login Credentials']);
+        }
+
+        $accessToken = Auth::user()->createToken('authToken')->accessToken;
+        return response(['user'=>Auth::user(), 'access_token' => $accessToken]);
```
[the remark this answers is not in the document]
Comment (surazad99) [reply backend/app/Http/Controllers/Api/LoginController.php:23]: jati rakhda ni huncha 200 to 300

File: backend/app/Http/Middleware/Authenticate.php [hunk 3]
```diff
@@ -15,7 +15,7 @@ class Authenticate extends Middleware
     protected function redirectTo($request)
     {
         if (! $request->expectsJson()) {
-            return route('login');
+            return route('error');
```
[the remark this answers is not in the document]
Comment (surazad99) [reply backend/app/Http/Middleware/Authenticate.php:18]: page ta taile banaune ho ni api bata information aaucha invalid token vanera ani taile login route ma pathaune ho frontend ma

File: backend/database/seeds/UserTableSeeder.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                    'role_id'=> 1,
+                    'phone' => 9841123456,
+                    'created_at' => \Carbon\Carbon::now(),
+                    'updated_at' => \Carbon\Carbon::now(),
+                ],
+            ];
+            DB::table('users')->insert($users);
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (surazad99) [reply backend/database/seeds/UserTableSeeder.php:35]: haha thanks

File: backend/routes/api.php [hunk 5]
```diff
@@ -14,6 +14,13 @@
 |
 */
 
-Route::middleware('auth:api')->get('/user', function (Request $request) {
-    return $request->user();
+
+
+//User
+Route::post('/login', 'Api\LoginController@login');
+Route::get('/error','Api\LoginController@error')->name('error');
```
[the remark this answers is not in the document]
Comment (surazad99) [reply backend/routes/api.php:21]: error dekhauna lai ho message ma invalid token aaucha esbata token milena vane

File: backend/app/Http/Middleware/Authenticate.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply backend/app/Http/Middleware/Authenticate.php:18]: same response ma error aauna parne ho ta ani

**Pull request merged**